src/shogun/mathematics/linalg/LinalgNamespace.h

```diff
@@ -98,6 +98,121 @@ LinalgBackendBase* infer_backend(const Container<T>& a, const Container<T>& b)
 }
 
 /**
+ * Transfers data to GPU memory. Does nothing if no GPU backend registered.
```

Comment:
I need to call `to_gpu` in the methods below. Or I can just declare them here but something needs to be in the front.